TimoCloud-Universal/src/main/java/cloud/timo/TimoCloud/base/managers/BaseInstanceManager.java

```diff
@@ -456,21 +449,42 @@ private void startProxy(BaseProxyObject proxy) {
         }
     }
 
-    private Integer getFreePort(int offset) {
+    private Integer getFreePortServer(BaseServerObject baseServerObject) throws ServerStartException {
+        Integer serverPort = getFreePortCommon(currentServerPort);
+
+        if (serverPort == null) {
+            TimoCloudBase.getInstance().severe("Error while starting server " + baseServerObject.getName() + ": No free port found. Please report this!");
+            throw new ServerStartException("No free port found");
+        }
+        serverPort++;
+        currentServerPort = serverPort;
+        if (currentServerPort.equals(maxServerPort)) currentServerPort = startPortServer;
```

Comment:
Remove these lines and do that in `getFreePortCommon` instead, as described below

Reply:
How should I remove these lines? I have to see if it is a proxy or server